lib/ossert/fetch/stackoverflow.rb

```diff
@@ -1,67 +1,165 @@
+      # Maximum number of pages to be fetched for a single project
+      MAX_PAGES = 50
+
+      # Create StackOverflow fetcher
+      # @param project [Project] a project to fill
       def initialize(project)
         @client = SimpleClient.new(BASE_URL)
         @project = project
         @cmn_params = SEARCH_PARAMS.merge(q: "\"#{project.name}\"")
-        @cmn_params[:key] = ENV['SO_TOKEN'] if ENV.has_key? 'SO_TOKEN'
+        @cmn_params[:key] = ENV['SO_TOKEN'] if ENV.key? 'SO_TOKEN'
       end
 
-      # Public: Trigger fetching
+      # Trigger fetching
       #
-      # Returns nothing.
+      # Starts fetching of the data from StackOverflow.
+      # *NOTE:* it mutates the project object passed during the initialisation.
+      # @return [nil] Not specified.
       def process
         process_questions_total
+        process_last_year_questions
       end
 
       private
 
-      # Public: Fetch all related questions
-      #
-      # Returns a response Hash from SO.
+      # Fetch all questions
+      # @param params [Hash<String, String>] special params for the search
+      # @return [Hash<Symbol, Hash>] a response from SO
       def fetch_questions(params = {})
         @client.get(SEARCH_URL, @cmn_params.merge(params))
       end
 
-      # Public: Fetch resolved questions
-      #
-      # Returns a response Hash from SO.
-      def fetch_resolved_questions
-        fetch_questions accepted: true
+      # Fetch resolved questions
+      # @param params [Hash<String, String>] special params for the search
+      # @return [Hash<Symbol, Hash>] a response from SO
+      def fetch_resolved_questions(params = {})
+        fetch_questions(params.merge(accepted: true))
+      end
+
+      # Fetch several pages of questions seamlessly
+      # @param params [Hash<String, String>] special params for the search
+      # @yieldparam question [Hash<Symbol, Hash>] one of the fetched questions
+      # @return [nil] Not specified
+      def fetch_questions_batch(params = {})
+        page = 1
+        resp = fetch_questions(params.merge(page: page))
+        while resp[:items].length.positive? && page < MAX_PAGES
+          resp[:items].each { |question| yield question }
+          page += 1
+          resp = fetch_questions(params.merge(page: page))
+        end
       end
 
-      # Public: Fetch & process total stats
+      # Process total stats
       #
-      # Returns nothing.
+      # Process aggregated data for the project on StackOverflow.
+      # @return [nil] Not specified.
       def process_questions_total
-        community.total.questions_count = fetch_questions[:total]
-        community.total.questions_resolved_count = fetch_resolved_questions[:total]
+        community.total.questions_count = fetch_questions(filter: :total)[:total]
+        community.total.questions_resolved_count = fetch_resolved_questions(filter: :total)[:total]
+        if last_question = fetch_questions(order: :desc, pagesize: 5).dig(:items, 0)
```

Comment:
Let's setup ruby version which has dig method. Tests should be green to continue